ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaContentCache.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionschema.internal
+
+import com.amazon.ion.IonSymbol
+import com.amazon.ion.IonValue
+import com.amazon.ionschema.InvalidSchemaException
+
+/**
+ * A class that caches the uninterpreted content of a Schema document. This enables us to determine which types are
+ * declared in a schema prior to having a [SchemaInternal] instance for the schema, which is especially useful for
+ * checking the validity of a type reference without having to resolve the type reference to a [TypeInternal] instance.
+ */
+internal class SchemaContentCache constructor(
```

Comment:
(Question) Where is this `SchemaContentCache` used?  Is this used by `IonSchemaSystem` while resolving a schema?

Reply:
Yes, this will be used by the `IonSchemaSystem`, but it will be while loading the content of a schema. That part will come in a later PR.